packages/hardhat/test/integration-tests/crabv2/crab-hedging-otc.ts

```diff
@@ -1000,6 +1000,39 @@ describe("Crab V2 flashswap integration test: time based hedging", function () {
             // reverting this back to one percent
             await crabStrategyV2.connect(owner).setHedgePriceThreshold(BigNumber.from(10).pow(16).mul(1));
         });
+        it("order signed for a different contract i.e EIP 712 attack", async () => {
+            // set the time to 1 hr from prev hedge
+            await provider.send("evm_increaseTime", [84600 + 3600]);
+            const trader = random;
+
+            // Calculate new Delta and the trades to make
+            const toGet = ethers.utils.parseUnits("3.125");
+            const toSell = ethers.utils.parseUnits("1");
+
+            // make the approvals for the trade and prepare the trade
+            await wSqueeth.connect(trader).approve(crabStrategyV2.address, toGet);
+
+            const orderHash = {
+                bidId: 0,
+                trader: trader.address,
+                quantity: toSell,
+                price: 1,
+                isBuying: false,
+                expiry: (await provider.getBlock(await provider.getBlockNumber())).timestamp + 600,
+                nonce: await crabStrategyV2.nonces(trader.address),
```

Comment:
gonna need a new test when nonce pr merged

Reply:
nonce PR might not get merged depending on the feedback from MMs